hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -570,6 +570,10 @@ private void invokeStartExecution() {
                         responses);
 
         mc.setJobStatus(RUNNING);
+        if (mc.hasTimeout()) {
+            long remainingTime = mc.remainingTime(Clock.currentTimeMillis());
```

Comment:
why do you schedule timeout that late (after `InitExecutionOperation` completed on all nodes)?

Reply:
btw, if the timeout is exceeded when the job is resumed, it should be not resumed at all instead of starting it and killing it 1ms later

Reply:
also, in previous implemenation, timeout could terminate also a suspended job. Now it does not, if I see correctly.

Reply:
> why do you schedule timeout that late (after InitExecutionOperation completed on all nodes)?

1. The interval from the job submission to the job being reported as `RUNNING` is negligible for any reasonable timeout.
2. Starting the timeout just after setting the `RUNNING` status makes it more consistent. And there is a reason why we report `RUNNING` status so late: initializing the execution may fail on some nodes or the job termination may be requested just after initializing the execution. They are both addressed in the current implementation and we try to avoid glitchy status updates.

Reply:
Ad.1. If the job starts too slowly, it should be possible to abort it using timeout. If the job hangs in the initiliazation (eg. due to some network glitches), it will never be cancelled due to the timeout. So initialization time may not always be that short. As part of initialization we connect to Kafka or JDBC etc. 
Ad.2. Do you mean statuses like `NOT_RUNNING->STARTING->FAILED`? I think this is perfectly fine, some jobs may not start running or partially start, not only due to the timeout but also other errors (like connection errors). Job can be also manually cancelled when it is still STARTING. Jet handles such cases. 

I still do not like the idea of scheduling 1 ms timeout, if the remaining time is <=0. Canceling immediately is more explicit, but maybe more complicated.

Reply:
I see we used the "hack" with 1ms timeout before, so for now we can leave it to not complicate the code now.

Reply:
You are right, timeout should cover the initialization as well. I restored the original behavior by scheduling the timeout at the of `MasterJobContext.tryStartJob()`.